File: scripts/task_tested_with_compile_script/url-encoding/perl/url-encoding.pl
sub urlencode {
    my $s = shift;
    $s =~ s/([^-A-Za-z0-9_.!~*'() ])/sprintf("%%%02X", ord($1))/eg;
    $s =~ tr/ /+/;
    return $s;
}



for ( $b=0; $b < 10000000; $b = $b + 1 ) {
	urlencode('http://foo bar/' + i);
}

File: Scripts/laptop/url-encoding/visual-basic-.net/url-encoding.vb
Imports System
Imports System.Web

Namespace Rextester
    Public Module Program
        Public Sub Main(args() As string)
	   For value As Integer = 0 To 10000000
            Dim encodedUrl As String = HttpUtility.UrlEncode("http%3A%2F%2Ffoo%20bar%2F" + value.ToString())
   	   'Console.WriteLine(encodedUrl)
           Next
	End Sub
    End Module
End Namespace
